api/dp-openapi.yaml

```diff
@@ -678,6 +678,474 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/checkout:
+    put:
+      tags:
+      - project
+      description: Check out some pages for proofreading
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: projstate
+        in: query
+        description: Project state
+        required: true
+        schema:
+          type: string
```

Comment:
same here, maybe an enum of all the states type?

Reply:
We could create a reference also.
Although one can write a Swagger definition first and write the code afterwards, I wrote the code first and it does validate the project state against an array of all possible project states (the global variable $PROJECT_STATES_IN_ORDER). It would be possible to put these into an enum in the Swagger file.
On the other hand, if the Swagger file is used by someone to write an application which uses the API they do not need to know what the states actually are since this value will be received from the application. Its only purpose is to ensure the project is still in the same state it was when they received the value.